crates/shielder-relayer/src/token.rs

```diff
@@ -0,0 +1,23 @@
+use alloy_primitives::Address;
+use rust_decimal::Decimal;
+use serde::{Deserialize, Serialize};
+
+#[derive(Copy, Clone, Debug, Default, Hash, Eq, PartialEq, Deserialize, Serialize)]
+pub enum TokenKind {
+    #[default]
+    Native,
+    ERC20(Address),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub enum PriceProvider {
+    Url(String),
+    Static(Decimal),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize, Serialize)]
+pub struct Token {
+    pub kind: TokenKind,
+    pub decimals: u32,
```

Comment:
This is ok, but imho decimals should be a responsibility of `TokenKind`, not `Token`:

```suggestion

    pub enum TokenKind {
        #[default]
        Native,
        ERC20(Address, u32),
    }

    pub struct Token {
        pub kind: TokenKind,
        pub price_provider: PriceProvider,
    }

    impl Token {
        fn decimals(&self) -> u32 {
            match self.kind {
                TokenKind::Native => 18,
                // bonus: you can actually fetch&cache ERC20 decimals
                // from the chain here, omitting configuring them manually
                TokenKind::ERC20(_, decimals) => decimals
            }
        }
    }
```

Reply:
1. you are assuming that every chain has 18 decimals for native - from yesterday discussion I got it that we shouldn't assume anything
2. for other parts of code it is easier to work with just information native/erc20, without additional config; in particular, decimal is only used for pricing; when the client makes a relay query, they shouldn't be bothered with specifying decimals for USDT; also, it'd be another check on our side if it is comatilbe with relayer config
3.

Reply:
I don't believe that we need native token to be configured, that's overkill. 

As for ERC-20, we actually don't need to configure anything but `token_address` and `price_feed`, as all other information is retrieve-able from chain (client's don't need to specify anything).

Imho the less you configure the better, but feel free to leave it as it is.

Reply:
done as you wish